src/Compiler/Service/TransparentCompiler.fs

```diff
@@ -291,6 +291,8 @@ type internal CompilerCaches(sizeFactor: int) =
 
     member val ItemKeyStore = AsyncMemoize(sf, 2 * sf, name = "ItemKeyStore")
 
+    member val ScriptClosure = AsyncMemoize(sf, 2 * sf, name = "ScriptClosure")
```

Comment:
Since it is enabled in long running scenarios, how much of memory impact we are talking about here? Service is already allocating a lot, we must be considerate.

Reply:
To give some numbers, I did the following:
- Hardcoded fsi to use an InteractiveChecker with `useTransparentCompiler = true`
- Started the modified fsi:
- Took Snapshot 1
- executed: `#r "nuget: FsHttp";;` 
- Took Snapshot 2
- executed: `#r "nuget: Fantomas.Core";;` 
- Took Snapshot 3

<img width="1724" alt="Screenshot 2024-02-06 115248" src="https://github.com/dotnet/fsharp/assets/3221269/293aa695-b60d-4acc-9f26-1f32a3112ec6">

![Screenshot 2024-02-06 115659](https://github.com/dotnet/fsharp/assets/3221269/e7110182-b732-4f69-982f-6f4572bf8c06)

<img width="1721" alt="Screenshot 2024-02-06 120030" src="https://github.com/dotnet/fsharp/assets/3221269/85307cc0-6191-4d45-8692-4696542a50e7">

It doesn't seem excessive to me but let me know if these numbers are of any use to you or if you want to have other scenarios profiled.

Reply:
That looks good, thanks!